board/bootstub.c

```diff
@@ -56,9 +56,9 @@ int main(void) {
   SHA_hash(&_app_start[1], len-4, digest);
 
   // verify version, last bytes in the signed area
-  uint32_t vers[2] = {0};
+  uint32_t vers[3] = {0};
   memcpy(&vers, ((void*)&_app_start[0]) + len - sizeof(vers), sizeof(vers));
-  if (vers[0] != VERS_TAG || vers[1] < MIN_VERSION) {
+  if (vers[0] != MCU_IDCODE || vers[1] != VERS_TAG || vers[2] < MIN_VERSION) {
```

Comment:
we should make a test for each of these cases, e.g. flash pedal fw to black panda, F4 fw to a red panda, etc.

Reply:
added test for flashing F4 with H7 app and H7 with F4.

Reply:
added flashing with F2 for both